blivet/fstab.py

```diff
@@ -0,0 +1,792 @@
+            clone._table.intro_comment = self._table.intro_comment
+        if self._table.trailing_comment is not None:
+            clone._table.trailing_comment = self._table.trailing_comment
+
+        entry = self._table.next_fs()
+        entries = []
+        while entry:
+            entries.append(entry)
+            entry = self._table.next_fs()
+
+        for entry in entries:
+            new_entry = self._copy_fs_entry(entry)
+            clone._table.add_fs(new_entry)
+
+        return clone
+
+    def __repr__(self):
+        entry = self._table.next_fs()
+        entries_str = ""
+        while entry:
+            entries_str += repr(FSTabEntry(entry=entry)) + '\n'
+            entry = self._table.next_fs()
+        return entries_str
+
+    def __iter__(self):
+        return FSTabManagerIterator(self)
+
+    def _copy_fs_entry(self, entry):
+        """ Create copy of libmount.Fs()
+        """
+        # Nope, it has to be done like this. Oh well...
+        new_entry = Fs()
+        new_entry.source = entry.source
+        new_entry.target = entry.target
+        new_entry.fstype = entry.fstype
+        new_entry.append_options(entry.options)
+        new_entry.freq = entry.freq
+        new_entry.passno = entry.passno
+        if entry.comment is not None:
+            new_entry.comment = entry.comment
+        return new_entry
+
+    def _parser_errcb(self, tb, fname, line):  # pylint: disable=unused-argument
+        """ Libmount interface error reporting function
+        """
+        log.error("Fstab parse error '%s:%s'", fname, line)
+        return 1
+
+    def entry_from_device(self, device):
+        """ Generate FSTabEntry object based on given blivet Device
+
+            :keyword device: device to process
+            :type device: :class: `blivet.devices.StorageDevice`
+            :returns: fstab entry object based on device
+            :rtype: :class: `FSTabEntry`
+        """
+
+        entry = FSTabEntry()
+
+        if device.format.uuid:
+            entry.spec = "UUID=%s" % device.format.uuid
+        else:
+            entry.spec = getattr(device, "fstab_spec", None)
+
+        entry.file = None
+        if device.format.mountable:
+            entry.file = device.format.mountpoint
+        if device.format.type == "swap":
+            entry.file = "swap"
+
+        entry.vfstype = device.format.type
+
+        return entry
+
+    def read(self, src_file=""):
+        """ Read the fstab file from path stored in self.src_file. Setting src_file parameter overrides
+            it with new value.
+
+            :keyword src_file: When set, reads fstab from path specified in it
+            :type src_file: str
+        """
+
+        # Reset table
+        self._table = Table()
+        self._table.enable_comments(True)
+
+        # resolve which file to read
+        if src_file == "":
+            if self.src_file is None:
+                # No parameter given, no internal value
```

Comment:
So if no `src_file` is given the `read` call is no-op? Shouldn't that be error?

Reply:
It is not a no-op. If no `src_file` is given it behaves like it is asked to read an empty file and (re)sets its internal table.

Reply:
It is not a no-op. Without set `src_file` the `read` method (re)sets its internal table. Same result as reading an empty fstab file.